ocaml/xapi-aux/env_record.mli

```diff
@@ -0,0 +1,27 @@
+(*
+ * Copyright (C) 2023 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+type t
+
```

Comment:
This needs documentation what is being constructed here.

Reply:
The way I understand it, the documentation could be something like this:

A Unix environment variable is a pair of a name and a value, both of which are strings. This module helps to construct a complex environment value that follows conventions specific conventions.

A value of type [t] represents a value. It is constructed recursively from OCaml values using functions like [str], [option], or [list]. An environment value of type [t] can be observed using  [to_shell_string].

I am not sure the above is correct. I would have modeled this in a way that captures the name of an environment variable together with its value but I don't see that, which adds to my confusion.

Reply:
@edwintorok Created the first draft of env_record so he might be able to explain it a bit better but I think the reason that the environment variable isn't captured with its value is because it will always be a simple string. It doesn't need any processing on it like quoting or turning it into a comma-separated list like the environment values do.

Reply:
https://github.com/xapi-project/xen-api/pull/5345/commits/6cae17d7935b39dc98dad5273fb85fa5922eeb20. I tried to summaries what I understood from @snwoods and @edwintorok's first draft.